beacon-chain/core/helpers/validators.go

```diff
@@ -271,13 +296,18 @@ func BeaconProposerIndex(ctx context.Context, state state.ReadOnlyBeaconState) (
 			return 0, err
 		}
 		if r != nil && !bytes.Equal(r, params.BeaconConfig().ZeroHash[:]) {
-			proposerIndices, ok := proposerIndicesCache.ProposerIndices(wantedEpoch+1, bytesutil.ToBytes32(r))
-			if ok {
-				return proposerIndices[state.Slot()%params.BeaconConfig().SlotsPerEpoch], nil
+			pid, err := cachedProposerIndexAtSlot(slot, [32]byte(r))
+			if err == nil {
```

Comment:
Given that you never use `errProposerIndexMiss` in any special way (e.g. `errors.Is`), returning a boolean as an indication of success seems more natural.

Reply:
The cache will be used by Kasey in the blob verification routine, I think it's safer to use the error, if it ends up being not used then we can remove it.